### PR TITLE
Add support neural query type

### PR DESCRIPTION
**Pull request by GranT1337**
### Description
Adds support for neural query type for [neural text search](https://opensearch.org/docs/latest/search-plugins/neural-text-search/)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — DISMISSED** [review]
Looks good! Do you think yo could add a doc and a working sample into  https://github.com/opensearch-project/opensearch-java/tree/main/guides and https://github.com/opensearch-project/opensearch-java/tree/main/samples as part of this?

**Comment by dblock** [comment]
@GranT1337 Looks like you'll need to fix spotless and such ... I'll change this whole CI doesn't run on first time committer nonsense

**Comment by reta** [comment]
@GranT1337 LGTM as well, please add changelog entry, thank you

**Comment by GranT1337** [comment]
> Looks good! Do you think yo could add a doc and a working sample into https://github.com/opensearch-project/opensearch-java/tree/main/guides and https://github.com/opensearch-project/opensearch-java/tree/main/samples as part of this?

@dblock Creating a sample is quite difficult. To make a simple example, you need to:

1. Upload the ML model into a cluster.
2. Deploy the ML model.
3. Define an ingest pipeline to create embeddings on the fly.
4. Create an index to store vector data.
5.  Use the uploaded model for search.

The model is quite large for a simple example, and opensearch-java currently does not allow performing these steps through Java code, i.e., sending JSON text is required.

**Review by dblock — DISMISSED** [review]
I'm good with this code. I think working samples and documentation are worth it, even if difficult or has a bunch of prerequisites, but I won't hold this PR for it.

File: java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/NeuralQuery.java
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/NeuralQuery.java:51]:
```suggestion
        return Kind.Neural;
```

File: java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/NeuralQuery.java
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/NeuralQuery.java:117]: Only `modelId` is supposed to be nullable, others are required

File: java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/NeuralQuery.java
Comment (GranT1337) [reply java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/NeuralQuery.java]: fixed

File: java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/NeuralQuery.java
Comment (GranT1337) [reply java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/NeuralQuery.java]: fixed

**Review by reta — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Pull request merged**
